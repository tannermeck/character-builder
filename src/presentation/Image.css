.imageContainer {
    display: flex;
    flex-direction: column;
    align-items: center;
    min-height: 100vh;
}
.head {
    height: 100px;
    width: 150px;
    background-size:contain;
}