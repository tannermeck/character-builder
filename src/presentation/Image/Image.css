.imageContainer {
    display: flex;
    flex-direction: column;
    align-items: center;
    min-height: 100vh;
}
.head {
    height: 100px;
    width: 150px;
    background-size: contain;
    background-repeat: no-repeat;
    background-position:center;
}
.torso {
    height: 100px;
    width: 150px;
    background-size: contain;
    background-repeat: no-repeat;
    background-position: center;
}
.legs {
    height: 100px;
    width: 150px;
    background-size: contain;
    background-repeat: no-repeat;
    background-position: center;
}