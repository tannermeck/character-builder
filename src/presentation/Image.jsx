import './Image.css';

export default function Image({head, torso, legs}){
    return (
        <section className="imageContainer">
            <div className="head"
                style={{ backgroundImage: `url(./${head}-head.png)`, 
                height: 200,
                width: 250,
                backgroundSize: 'cover',
                backgroundRepeat: 'no-repeat'
            }}>
            </div>
            <div className="torso"
                style={{ backgroundImage: `url(./${torso}-torso.png)`, 
                height: 200,
                width: 250,
                backgroundSize: 'cover',
                backgroundRepeat: 'no-repeat'
            }}>
            </div>
            <div className="legs"
                style={{ backgroundImage: `url(./${legs}-legs.png)`, 
                height: 200,
                width: 250,
                backgroundSize: 'cover',
                backgroundRepeat: 'no-repeat'
            }}>
            </div>
        </section>
    )
}