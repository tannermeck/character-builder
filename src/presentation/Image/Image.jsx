import './Image.css';

export default function Image({head, torso, legs}){
    return (
        <section className="imageContainer">
            <div className="head"
                style={{ backgroundImage: `url(./${head}-head.png)` 
            }}>
            </div>
            <div className="torso"
                style={{ backgroundImage: `url(./${torso}-torso.png)`
            }}>
            </div>
            <div className="legs"
                style={{ backgroundImage: `url(./${legs}-legs.png)`
            }}>
            </div>
        </section>
    )
}