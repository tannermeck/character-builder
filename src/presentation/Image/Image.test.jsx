import { render } from '@testing-library/react';
import Image from './Image';

it('should render images in a snapshot', () => {
    const { container } = render(<Image head={'goat'} torso={'black-blouse'} legs={'speedo'} />)
    expect(container).toMatchSnapshot();
    
})