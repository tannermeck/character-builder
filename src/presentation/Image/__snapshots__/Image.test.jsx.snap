// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render images in a snapshot 1`] = `
<div>
  <section
    class="imageContainer"
  >
    <div
      class="head"
      style="background-image: url(./goat-head.png);"
    />
    <div
      class="torso"
      style="background-image: url(./black-blouse-torso.png);"
    />
    <div
      class="legs"
      style="background-image: url(./speedo-legs.png);"
    />
  </section>
</div>
`;
